### PR TITLE
1.17>1.18 parity (sort of)

### PR DESCRIPTION
**Pull request by xanthian**
Fixes grinder test build 

Also implements following 1.18 features :
Textures for 1.18 generators by @xanthian 
Lighting particle fix by @xanthian 
Rotational tank textures by @xanthian 
Drain breaking fix by @modmuss50 
waterlogged cable cover fix by @modmuss50 
Ore breaking fix by @equal-l2 
Chainsaw fix by @equal-l2

### CODE REVIEW

File: src/gametest/groovy/techreborn/test/machine/GrinderTest.groovy [hunk 1]
```diff
@@ -32,7 +32,7 @@ import techreborn.test.TRGameTest
 import techreborn.test.TRTestContext
 
 class GrinderTest extends TRGameTest {
-    @GameTest(structureName = EMPTY_STRUCTURE, tickLimit = 150)
+    @GameTest(structureName = fabric-gametest-api-v1:empty, tickLimit = 150)
```
Comment (modmuss50) [inline src/gametest/groovy/techreborn/test/machine/GrinderTest.groovy:35]: I dont believe this is needed as it was caused by a bug in the version of groovy the 1.18 branch is using.

Comment (xanthian) [reply src/gametest/groovy/techreborn/test/machine/GrinderTest.groovy:35]: couldn't get it to build without adding this, failed immediately even with just the 1.17 master, no changes.
(locally, using gradlew build, jdk17)
this bit is kinda over my head tbh :P i just flailed around until it worked

Comment (modmuss50) [reply src/gametest/groovy/techreborn/test/machine/GrinderTest.groovy:35]: Humm odd, leave it then. I didn’t bother to look into it any more when porting to 1.18

Comment (xanthian) [reply src/gametest/groovy/techreborn/test/machine/GrinderTest.groovy:35]: build failed on git, but been playing it all evening without any issues.

**Comment by xanthian** [comment]
closed this PR as it screwed up and couldnt amend it.

**Pull request closed without merging**